### ISSUE TITLE
是否应该用paddle.randint取代random.randint？

### ISSUE THREAD

**Issue by Wong4j** [PaddlePaddle/Paddle#68055]
### bug描述 Describe the Bug

我在使用`paddle.vision.transforms.RandomCrop`的时候，固定seed，但crop的结果仍然是随机的。发现这是因为 [transforms.py](https://github.com/PaddlePaddle/Paddle/blob/99cf4b14bc252a727b5ab17a99455c620ea20391/python/paddle/vision/transforms/transforms.py#L1407)中用了random.randint来生成随机数。
一般用户只会记得去设置paddle.seed和np.random.seed，并不知道Layer内部调用了random。所以建议用paddle来生成随机数。

**Comment by zhwesky2010** [comment]
@HydrogenSulfate 辛苦看下这个API问题

**Comment by HydrogenSulfate** [comment]
> 一般用户只会记得去设置paddle.seed和np.random.seed，并不知道Layer内部调用了random。所以建议用paddle来生成随机数。

1. 随机数固定过程一般至少需要固定深度学习框架的seed、numpy、random三个库的随机数，无论是paddle、pytorch或者其他深度学习框架都建议这么做，这与paddle官网上的文档和[谷歌搜索的结果](https://wandb.ai/sauravmaheshkar/RSNA-MICCAI/reports/How-to-Set-Random-Seeds-in-PyTorch-and-Tensorflow--VmlldzoxMDA2MDQy)一致：
![image](https://github.com/user-attachments/assets/3784adb1-bc8e-4d3b-80e2-521e5519b32f)，random本身也是一个随机库，不能忽略它

2. 使用`paddle.randint`至少会引起两个问题：不必要的调用开销（kernel调用、Tensor转回numpy的拷贝），产生的随机序列会发生变化，导致不可预知的影响。因此根据实际情况，动态图的transform操作对象是numpy，并不需要Tensor的参与，因此使用random或numpy生成随机数是更合理的。
